### PR TITLE
Add support for the crawling animation.

### PR DESCRIPTION
**Pull request by ShakyTimon**
Hey, in my commit I added support for the crawling animation in Minecraft, and considering there is not native methods to check for this, I used the process of elimination, and the height of your hitbox. If you're not sneaking, swimming, or gliding, but have a reduced hitbox height, you are 'crawling'.

I struggled to build WeaponMechanics due to authentication being required for a dependency:
```> Could not get resource 'https://maven.pkg.github.com/WeaponMechanics/MechanicsAutoDownload/me/cjcrafter/mechanicsautodownload/1.3.1/mechanicsautodownload-1.3.1.pom'. Received status code 401 from server: Unauthorized.```

If this is an issue on my side, I would much appreciate some help.

I did however test what I added to EntityWrapper inside my own plugin, source:

```
    @Override
    public boolean onCommand(CommandSender commandSender, org.bukkit.command.Command command, String s, String[] strings) {
        Player player = (Player) commandSender;

        if (player.isSneaking()) {
            player.sendMessage("You are sneaking.");
            return true;
        }
        if (player.isSwimming()) {
            player.sendMessage("You are swimming.");
            return true;
        }
        if (player.isGliding()) {
            player.sendMessage("You are gliding.");
            return true;
        }

        final double crawlingMaxHeight = 1.5;
        double hitboxHeight = player.getBoundingBox().getHeight();
        if (hitboxHeight <= crawlingMaxHeight) {
            player.sendMessage("You are crawling.");
        } else {
            player.sendMessage("You are not crawling.");
        }
        return true;
    }
```
Here's what that looked like: https://cdn.discordapp.com/attachments/1128569715976065054/1191678119753560124/170418859513897991.mp4?ex=65a64fbc&is=6593dabc&hm=cd1e203f954fd4a1deab397fae174cf05eda80b8e0122e5b9709e70a2fe196c1&

This is my first PR to a large project so I hope that everything is okay,

Seems that some of the scripts inside the lib folder which I edited as I am on linux were included into the PR, those are not meant to be there haha, so if this gets merged, please do not merge those, and if they cannot be excluded I can redo everything and make a new PR.

zunker says hi

### CODE REVIEW

**Review by CJCrafter — CHANGES_REQUESTED** [review]
Just a minor change

Thanks very much for contributing! :)

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/wrappers/EntityWrapper.java [hunk 1]
```diff
@@ -162,6 +162,30 @@ public boolean isSneaking() {
         return false;
     }
 
+    /**
+     * Returns <code>true</code> if the player is 'crawling'.
+     *
+     * @return <code>true</code> when the player is 'crawling'.
+     */
+    public boolean isCrawling() {
```
[the remark this answers is not in the document]
Comment (ShakyTimon) [reply WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/wrappers/EntityWrapper.java]: I've implemented it, please check, and if everything is okay, merge it.

As for the scripts, please don't include those in the merge, they were very buggy, untested, and 'quick' rewrites.

**Comment by CJCrafter** [comment]
You should revert those script changes then

**Comment by ShakyTimon** [comment]
> You should revert those script changes then

After some tinkering with git; done.

**Review by CJCrafter — APPROVED** [review]

**Comment by CJCrafter** [comment]
Thanks for contributing :+1:

**Pull request merged**